### PR TITLE
PR_Revucky

### PR DESCRIPTION
**Pull request by samuelrevucky**
- simplified reading from the input file and creating an empty matrix
- sanitized inputs

### CODE REVIEW

File: ConcatenateMatrices.java [hunk 1]
```diff
@@ -3,33 +3,69 @@
...
+            return;
+        }
+
+        int m, n;
+        try {
+            m = scanner.nextInt();
+            n = scanner.nextInt();
+            if(m < 1 || n < 1) throw new IllegalArgumentException();
+        }
+        catch (Exception e){
```
Comment (janmazak) [inline ConcatenateMatrices.java:28]: Catching everything is generally considered a bad practice. 
https://softwareengineering.stackexchange.com/questions/164256/is-catching-general-exceptions-really-a-bad-thing

File: ConcatenateMatrices.java [hunk 2]
```diff
@@ -3,33 +3,69 @@
...
+            if(scanner != null) scanner.close();
+            if(output != null) output.close();
+            return;
+        }
+
+        int m, n;
+        try {
+            m = scanner.nextInt();
+            n = scanner.nextInt();
+            if(m < 1 || n < 1) throw new IllegalArgumentException();
```
Comment (janmazak) [inline ConcatenateMatrices.java:26]: It would be nice to provide an error message. (Very useful for debugging, too: sometimes it is not possible to log stacktrace properly, so the message is the only indication of the source of the exception.)

File: ConcatenateMatrices.java [hunk 3]
```diff
@@ -3,33 +3,69 @@
...
+            output = new PrintStream("vystup.txt");
+        }
+        catch (Exception e){
+            System.out.println("Problem s nacitanim/vytvorenim vstupneho/vystupneho suboru.");
+            if(scanner != null) scanner.close();
+            if(output != null) output.close();
+            return;
+        }
+
+        int m, n;
```
Comment (janmazak) [inline ConcatenateMatrices.java:21]: In general, using non-initialized variables (they are initialized to zero, but that is a meaningless value) has drawbacks. If someone later adds code between lines 21 and 22, it might introduce a bug. (Such a scenario is too far-fatched in this trivial example, but not every compiler is good at reporting such potentially dangerous use, consider e.g. javascript.)

This problem would entirely disappear if the code was split into three separate parts:
(1) reading and parsing input;
(2) validating input (e.g. positive values for m, n);
(3) subsequent computations on valid input.

File: ConcatenateMatrices.java [hunk 4]
```diff
@@ -3,37 +3,70 @@
...
-        String[] rozmery = vstup.split(" ");
-        int m = Integer.parseInt(rozmery[0]);
-        int n = Integer.parseInt(rozmery[1]);
+    public static void main(String[] args){
 
-        String[][] matrix = new String[m][]; //matica a jej vynulovanie
-        for (int k = 0; k < m; k++) {
-            matrix[k] = new String[n];
+        Scanner scanner = null;
+        PrintStream output = null;
```
Comment (janmazak) [inline ConcatenateMatrices.java:9]: Initializing variables to meaningless values is better avoided. See the comment below.

File: ConcatenateMatrices.java [hunk 5]
```diff
@@ -3,37 +3,70 @@
...
             }
         }
+
         while (scanner.hasNextLine()) { //cita pokial nenarazi na koniec (neexistujuci prvy riadok matice)
-            for (int i = 0; i < m; i++) { //ak je dalsi riadok, tak urcite bude m-riadkov matice
+            for (int i = 0; i < m; i++) {
+                if(!scanner.hasNextLine()){
+                    System.out.println("Nedostatocny pocet zadanych riadkov matice.");
+                    if(scanner != null) scanner.close();
+                    if(output != null) output.close();
```
Comment (janmazak) [inline ConcatenateMatrices.java:47]: Doing the same test several times is definitely not the way to go. Such corner cases should be dealt with upfront to avoid polluting the code. Read about happy paths:
https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88

File: ConcatenateMatrices.java [hunk 6]
```diff
@@ -3,37 +3,70 @@
...
             for (int j = 0; j < n; j++) {
                 matrix[i][j] = "";
             }
         }
+
         while (scanner.hasNextLine()) { //cita pokial nenarazi na koniec (neexistujuci prvy riadok matice)
-            for (int i = 0; i < m; i++) { //ak je dalsi riadok, tak urcite bude m-riadkov matice
+            for (int i = 0; i < m; i++) {
+                if(!scanner.hasNextLine()){
+                    System.out.println("Nedostatocny pocet zadanych riadkov matice.");
```
Comment (janmazak) [inline ConcatenateMatrices.java:45]: It would probably be better to throw an exception on invalid input, wrap reading the input in a function and when calling the function catch those exceptions in a simple handler.

**Review by janmazak — COMMENTED** [review]
The commit structure is nice, though there is a lot to fix yet. Let's say I give you 8 pts. You might want to finish the task later.
